dfc/proxy.go

```diff
@@ -607,7 +607,8 @@ func (p *proxyrunner) httpobjput(w http.ResponseWriter, r *http.Request) {
 		p.invalmsghdlr(w, r, errstr)
 		return
 	}
-	redirecturl := fmt.Sprintf("%s%s?%s=%t", si.DirectURL, r.URL.Path, URLParamLocal, p.islocalBucket(bucket))
+	redirecturl := fmt.Sprintf("%s%s?%s=%t&%s=%s", si.DirectURL, r.URL.Path, URLParamLocal,
+		p.islocalBucket(bucket), URLParamDaemonID, p.httprunner.si.DaemonID)
```

Comment:
what about other places that call target put? like rebalancing? I don't know the answer myself, just asking.

Reply:
Good question - that case is handled here: https://github.com/NVIDIA/dfcpub/pull/43/files#diff-55511de5ecebac432bae78861a0686a5R477